ouroboros-network/src/Ouroboros/Network/PeerSelection/PeerSelectionActions.hs

```diff
@@ -69,8 +69,10 @@ data PeerSelectionActionsArgs peeraddr peerconn exception m = PeerSelectionActio
   psReadPeerSharingController :: STM m (Map peeraddr (PeerSharingController peeraddr m)),
   -- ^ peer sharing registry
   psUpdateOutboundConnectionsState
-                              :: OutboundConnectionsState -> STM m ()
+                              :: OutboundConnectionsState -> STM m (),
   -- ^ Callback which updates information about outbound connections state.
+  psChurnMutex                :: StrictTMVar m ()
+  -- ^ this is used to coordinate the actions of peer selection and churn governor
```

Comment:
I suggested before to use something like `withMVar`, we cannot use `MVar` since we need `tryReadTMVar` in the `Monitor.targets` actions.  We should define our own `Lock` data type with it's API for this purpose.  I based it on the implementation of `withMVar`:
```hs
module Control.Concurrent.Class.MonadSTM.Lock
  ( Lock
  , newLock
  , withLock
  )
  where

newtype Lock m = Lock (TMVar m ())

newLock :: MonadSTM m => Lock m
newLock = Lock <$> newTMVarIO ()

withLock :: MonadSTM m => Lock m -> m a -> m a
withLock (Lock m) =
      mask $ \restore -> do
      a <- atomically $ takeTMVar m
      b <- restore (io a) `onException` atomically (putTMVar m a)
      atomically (putMVar m a)
      return b
      
isLocked :: Lock m -> STM m Bool
isLocked (Lock m) = isNothing <$> tryReadTMVar m
```